kolibri/plugins/learn/assets/src/views/TopicsPage.vue

```diff
@@ -527,12 +536,21 @@
         return this.channel.name;
       },
       resources() {
-        const resources = this.contents.filter(content => content.kind !== ContentNodeKinds.TOPIC);
-        // If there are no topics, then just display all resources we have loaded.
-        if (!this.topics.length) {
-          return resources;
+        return this.contents.filter(content => content.kind !== ContentNodeKinds.TOPIC);
+      },
+      resourcesDisplayed() {
+        let displayed;
+        // if no folders are shown at this level, show more resources to fill the space
+        if (!this.showMoreResources) {
+          displayed = this.resources.slice(0, this.childrenToDisplay);
+          // otherwise display all resources
+        } else {
+          displayed = this.resources;
         }
-        return resources.slice(0, this.childrenToDisplay);
+        return displayed;
+      },
```

Comment:
This is missing the topics length check now, and the comments are somewhat misleading.

Maybe something like this? (using a return early pattern to avoid having to assign a variable):
```
      resourcesDisplayed() {
        // if no folders are shown at this level, show more resources to fill the space
        // or if the user has explicitly requested to show more resources
        if (!this.topics.length || this.showMoreResources) {
          return this.resources;
        } 
        return this.resources.slice(0, this.childrenToDisplay);
```